File: src/aimarafone/objects/suit.py
from enum import Enum


class Suit(Enum):
    """Enum representing card suits."""

    BASTONI = "Bastoni"
    DENARA = "Denara"
    SPADE = "Spade"
    COPPE = "Coppe"


if __name__ == "__main__":
    for suit in Suit:
        print(suit.name, suit.value)

File: android/app/src/main/java/com/maraffa/beccaccino/data/model/Suit.kt
package com.maraffa.beccaccino.data.model

enum class Suit(val displayName: String, val firebaseKey: String, val symbol: String) {
    COINS("Denari", "coins", "D"),
    CUPS("Coppe", "cups", "C"),
    SWORDS("Spade", "swords", "S"),
    CLUBS("Bastoni", "clubs", "B");

    companion object {
        fun fromFirebaseKey(key: String): Suit =
            values().first { it.firebaseKey == key }

        fun fromFirebaseKeyOrNull(key: String?): Suit? =
            if (key.isNullOrBlank()) null else values().firstOrNull { it.firebaseKey == key }
    }
}
